File: java/org/arch/cast/Legacy.java
/*public class MainActivity extends AppCompatActivity {

    private byte[] buffer;
    private DatagramSocket socket = null;
    private DatagramPacket packet;

    private AudioRecord recorder;
    private int sampleRate = 16000;
    private int channelConfig = AudioFormat.CHANNEL_IN_MONO;
    private int audioFormat = AudioFormat.ENCODING_PCM_8BIT;
    int size, minBufSize = AudioRecord.getMinBufferSize(sampleRate, channelConfig, audioFormat);
    boolean recording = false;

    @Override
    @SuppressLint("ClickableViewAccessibility")
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if(ContextCompat.checkSelfPermission(MainActivity.this, Manifest.permission.RECORD_AUDIO) == PackageManager.PERMISSION_DENIED)
            ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.RECORD_AUDIO},0);

        Button voice = findViewById(R.id.transmit);
        voice.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View view, MotionEvent event) {
                if(event.getAction() == MotionEvent.ACTION_DOWN) {
                    new Thread(){
                        public void run(){
                            try{
                                if(socket == null) {
                                    socket = new DatagramSocket();
                                    socket.connect(InetAddress.getByName("10.0.0.14"), 1238);
                                }
                                buffer = new byte[minBufSize];
                                recording = true;
                                recorder = new AudioRecord(MediaRecorder.AudioSource.MIC,sampleRate,channelConfig,audioFormat,minBufSize*16);
                                recorder.startRecording();
                                while(recording) {
                                    size = recorder.read(buffer, 0, buffer.length);
                                    packet = new DatagramPacket(buffer, size);
                                    socket.send(packet);
                                }
                            }
                            catch(SocketException se){
                                Log.e("Socket","Socket Exception!");
                            }
                            catch(UnknownHostException uhe){
                                Log.e("Socket","Unknown Host Exception!");
                            }
                            catch(IOException ioe){
                                Log.e("Socket","IO Exception!");
                            }
                        }
                    }.start();
                }
                else if(event.getAction() == MotionEvent.ACTION_UP)
                {
                    new Thread(){
                        public void run(){
                            recording = false;
                            recorder.release();
                        }
                    }.start();
                }
                return false;
            }
        });
    }
}
        new Thread() {
            @Override
            public void run() {
                while (metaConn) {
                    try {
                        op = queue.poll(Long.MAX_VALUE, TimeUnit.SECONDS);
                        stream.writeInt(op);
                        if (op == 0) {
                            stream.writeInt(queue.poll(Long.MAX_VALUE, TimeUnit.SECONDS));
                            stream.writeInt(queue.poll(Long.MAX_VALUE, TimeUnit.SECONDS));
                        }
                    } catch (Exception exception) {
                        metaConn = false;
                        startActivity(new Intent(MainActivity.this, LaunchActivity.class));
                        MainActivity.this.finish();
                    }
                }
            }
        }.start();
    }

*/

File: java/org/arch/cast/Legacy.kt
/*package org.arch.cast

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

import java.io.DataInputStream
import java.io.DataOutputStream
import java.net.DatagramPacket
import java.net.InetSocketAddress
import java.net.Socket
import java.util.concurrent.LinkedBlockingQueue

class MainActivity : AppCompatActivity() {

    internal val ipAddr = "10.0.0.11"
    internal val tcpPort = 1238
    internal val udpPort = 1873

    internal var uuid = ByteArray(16)
    internal var channel = 0
    internal var metaConn = false
    internal var metaInit = false
    internal var dataConn = false
    internal var metaQueue = LinkedBlockingQueue<Int>()
    internal var dataInputQueue = LinkedBlockingQueue<DatagramPacket>()
    internal var dataOutputQueue = LinkedBlockingQueue<DatagramPacket>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        StreamThread().start()
    }

    internal inner class StreamThread : Thread() {
        override fun run() {
            var stream : Socket
            var input : DataInputStream
            var output : DataOutputStream

            while (!metaConn) {
                try {
                    stream = Socket()
                    stream.connect(InetSocketAddress(ipAddr, tcpPort), 2000)
                    input = DataInputStream(stream.getInputStream())
                    output = DataOutputStream(stream.getOutputStream())
                    metaConn = true

                    println("Connected!")
                    runOnUiThread {
                        textChannel.isEnabled = true
                    }
                } catch (exception: Exception) {
                    println("Failed!")
                }
            }
        }
    }

    internal inner class DataInputThread : Thread() {
        override fun run() {}
    }

    internal inner class DataOutputThread : Thread() {
        override fun run() {}
    }
}
*/